nengo/utils/lock.py

```diff
@@ -0,0 +1,51 @@
+import errno
+import os
+import os.path
+import time
+
+
+class TimeoutError(Exception):
```

Comment:
Should this be a `NengoException`? At least in the cache this kind of exception should always be caught and handled.